tools/istio-iptables/pkg/capture/helper.go

```diff
@@ -0,0 +1,173 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package capture
+
+import (
+	"strings"
+
+	istiolog "istio.io/istio/pkg/log"
+	"istio.io/istio/tools/istio-iptables/pkg/builder"
+	"istio.io/istio/tools/istio-iptables/pkg/constants"
+	dep "istio.io/istio/tools/istio-iptables/pkg/dependencies"
+)
+
+func CombineMatchers(values []string, matcher func(value string) []string) []string {
+	matchers := make([][]string, 0, len(values))
+	for _, value := range values {
+		matchers = append(matchers, matcher(value))
+	}
+	return Flatten(matchers...)
+}
+
+func Flatten(lists ...[]string) []string {
+	var result []string
+	for _, list := range lists {
+		result = append(result, list...)
+	}
+	return result
+}
+
+// VerifyIptablesState function verifies the current iptables state against the expected state.
+// The current state is considered equal to the expected state if the following three conditions are met:
+//   - Every ISTIO_* chain in the expected state must also exist in the current state.
+//   - Every ISTIO_* chain must have the same number of elements in both the current and expected state.
+//   - Every rule in the expected state (whether it is in an ISTIO or non-ISTIO chain) must also exist in the current state.
+//     The verification is performed by using "iptables -C" on the rule produced by our iptables builder. No comparison of the parsed rules is done.
+//
+// Note: The order of the rules is not checked and is not used to determine the equivalence of the two states.
```

Comment:
```suggestion
// Note: The order of the rules is not checked and is not used to determine the equivalence of the two states.
```

Remind me again why we don't check this?

Reply:
This limitation arises from using iptables -C as part of the logic. The iptables -C command only verifies that the rule exists but does not check its position. To verify the position, we would need to compare the expected state with the current state. However, since the current state can be altered during the round trip to the kernel, this is pretty much prone to breakage.

Reply:
That's right, ty. We should make sure this is clearly documented for future generations.